ir/src/symbol_table.rs

```diff
@@ -241,6 +257,23 @@ impl SymbolTable {
                     ))
                 }
             }
+            IdentifierType::TransitionVariable(transition_variable) => {
+                if let TransitionVariableType::Vector(vector) = transition_variable.value() {
+                    if vector_access.idx() < vector.len() {
+                        Ok(symbol_type)
+                    } else {
+                        Err(SemanticError::vector_access_out_of_bounds(
+                            vector_access,
+                            vector.len(),
+                        ))
+                    }
```

Comment:
This section seems to be a duplication of the code in the `Constant` case above. Could we use a helper function instead?